src/python/visclaw/frametools.py

```diff
@@ -176,22 +183,69 @@ def plot_frame(framesolns,plotdata,frameno=0,verbose=False):
 
                 # loop over patches:
                 # ----------------
-    
+
+                # Mask out covered coarse grid regions.
                 for stateno,state in enumerate(framesoln.states):
                     #print '+++ stateno = ',stateno
-                    state = framesoln.states[stateno]
                     patch = state.patch
 
+
+                    # From pyclaw/src/pyclaw/state.py : A 'state' is an array of
+                    # values (stored as [mq,i,j], mq = field.
+                    # self.q   = self.new_array(num_eqn)
+                    # self.aux = self.new_array(num_aux)
+
+
+                    patch = state.patch
+
+
                     current_data.add_attribute('patch',patch)
                     current_data.add_attribute("level",1)
                     current_data.add_attribute('q',state.q)
                     current_data.add_attribute('aux',state.aux)
                     current_data.add_attribute('xlower',patch.dimensions[0].lower)
                     current_data.add_attribute('xupper',patch.dimensions[0].upper)
-                    
+
                     current_data.add_attribute("x",patch.grid.p_centers[0])
                     current_data.add_attribute("dx",patch.delta[0])
 
+                    # -------------------------------------------------------------
```

Comment:
Was this added to support transparency for Google Earth?  If so I am not sure if this is how we would want to go about doing this.  At the very least maybe only adding this to the `current_data` dict if `use_for_kml` is set to `True` or compute it later.